Map block: Use the Mapbox access token directly if possible

Copons opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to to-test.md in a new commit as part of your PR. -->

Fixes #14475 

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->
* On WordPress.com, use the Mapbox access token directly (if it exists) instead of calling an endpoint.

As a direct effect, this saves WordPress.com Simple sites one unnecessary API call.
As a side effect, this allows _private_ WordPress.com Simple sites to bypass the unnecessary permissions checks and use the Mapbox access token as intended.

#### Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

This is a WordPress.com-only change, so on Jetpack just smoke test for regressions.

* Apply D38503  and D38506.
* Use a _private_ WordPress.com site that doesn't have a custom Mapbox access token saved (you can remove it from the Map block sidebar _before_ applying this PR).
* Insert a Map block, and make sure it doesn't ask for the access token anymore.

#### Proposed changelog entry for your changes:
<!-- Please do not leave this empty. If no changelog entry needed, state as such. -->
* N/A (WordPress.com-only change)

Copons committed:
Use the Mapbox access token directly if possible

apeatling commented:
I want to make sure this is correct, here's what happened when I tested:

* I applied both diffs to my sandbox, and sandboxed a private dotcom simple site, and public-api.
* I added a map block to a post, this worked correctly, the token was present.
* I removed the token from the block. The block still worked on the front end, and the key readded when I refreshed the editor.
* The key was present when adding a new map block to the page.
* Everything still worked as expected when using this in Jetpack locally.

creativecoder reviewed (changes requested):
This isn't working on the front end for me, when viewing a map within a post or page for private sites... the map is blank and does not load.

I think this is because the map block is still [making an api call for the key](https://github.com/Automattic/jetpack/blob/08b4f279d26287ae4f26cbc9aef60b57d0796d36/extensions/blocks/map/map.php#L22) when the map loads.

Ideally there would be a shared library function that is called both from the [map block](https://github.com/Automattic/jetpack/blob/08b4f279d26287ae4f26cbc9aef60b57d0796d36/extensions/blocks/map/map.php#L22) and from the [service-api-keys endpoint](https://github.com/Automattic/jetpack/blob/08b4f279d26287ae4f26cbc9aef60b57d0796d36/_inc/lib/core-api/wpcom-endpoints/service-api-keys.php#L294) that contains the logic for getting the mapbox key: either directly from a constant on wpcom or through the API on Atomic sites.

Copons committed:
Make sure the Mapbox access token is used on both front and back end

Copons committed:
Improve consistency

Copons commented:
@creativecoder I've updated this to make sure the front end does roughly the same attempts (and in the same order) to get an access token as the back end:

1. Custom token saved in site option.
2. WPCOM token via WPCOM constant.
3. Cached WPCOM token.
4. WPCOM token via API.

I decided against creating an external library because the actual overlap between the two implementations is not that much imho. 🤔

Copons commented:
> I want to make sure this is correct, here's what happened when I tested:
> 
> * I applied both diffs to my sandbox, and sandboxed a private dotcom simple site, and public-api.
> * I added a map block to a post, this worked correctly, the token was present.
> * I removed the token from the block. The block still worked on the front end, and the key readded when I refreshed the editor.
> * The key was present when adding a new map block to the page.
> * Everything still worked as expected when using this in Jetpack locally.

@apeatling I might be misunderstanding, but you shouldn't have seen any token UI.
See for example:

![Feb-05-2020 15-19-38](https://user-images.githubusercontent.com/2070010/73854797-f9809000-482a-11ea-926c-9e0fd69a965f.gif)

| WPCOM (no token UI) | Jetpack |
| - | - |
| <img width="274" alt="Screenshot 2020-02-05 at 15 20 35" src="https://user-images.githubusercontent.com/2070010/73854973-45333980-482b-11ea-826f-d28eddd7d7d5.png"> | <img width="272" alt="Screenshot 2020-02-05 at 15 21 09" src="https://user-images.githubusercontent.com/2070010/73854981-48c6c080-482b-11ea-877d-80271856b8d4.png"> |

creativecoder reviewed (approved):
The map load is now working for me with a sandboxed private Simple Site.

I'll approve this, but it still seems very odd to me that `jetpack_get_mapbox_api_key` makes an api call to `wpcom/v2/service-api-keys/mapbox` which then makes another api call to `/wpcom/v2/sites/' . $site_id . '/mapbox'`, rather than calling the second endpoint directly and avoiding the overhead of 2 separate api requests.

Refactoring `jetpack_get_mapbox_api_key` to

- Return the `mapbox_api_key` from the site option, if found, OR
- Return the wpcom key, if appropriate, OR
- Return the cached wpcom key, if available, OR
- Get a key from `/wpcom/v2/sites/' . $site_id . '/mapbox'`

and then using that function both from the `map.php` file for rendering the block on the front end and from the `wpcom/v2/service-api-keys/mapbox` for rendering the map in the editor would be much easier to follow and is worth doing, in my opinion.

Copons commented:
@creativecoder I agree with you.
I'll take the time today to see if I can group all my currently open and interdependent PRs and diffs into one single change, easier to test and merge.

Copons commented:
Replaced by #14597
